react/features/face-centering/middleware.js

```diff
@@ -0,0 +1,135 @@
+// @flow
+
+import {
+    CONFERENCE_JOINED,
+    CONFERENCE_WILL_LEAVE,
+    getCurrentConference
+} from '../base/conference';
+import { JitsiConferenceEvents } from '../base/lib-jitsi-meet';
+import { MiddlewareRegistry } from '../base/redux';
+import { TRACK_UPDATED, TRACK_ADDED, TRACK_REMOVED } from '../base/tracks';
+import { changeTrack, resetTrack } from '../facial-recognition';
+import { shouldDisplayTileView } from '../video-layout';
+import { SET_TILE_VIEW } from '../video-layout/actionTypes';
+import { VIRTUAL_BACKGROUND_TRACK_CHANGED } from '../virtual-background/actionTypes';
+
+import { UPDATE_FACE_COORDINATES } from './actionTypes';
+import {
+    loadWorker,
+    stopFaceRecognition,
+    startFaceRecognition
+} from './actions';
+import { FACE_BOX_EVENT_TYPE } from './constants';
+
+MiddlewareRegistry.register(({ dispatch, getState }) => next => action => {
+    const state = getState();
+    const { faceCoordinatesSharing } = state['features/base/config'];
+    const isInTileView = shouldDisplayTileView(state);
+
+    if (!getCurrentConference(getState())) {
+        return next(action);
+    }
+
+    if (action.type === CONFERENCE_JOINED) {
+        if (faceCoordinatesSharing?.enabled) {
+            dispatch(loadWorker());
+
+            if (isInTileView) {
```

Comment:
Why only for tile view? Wouldn't this also help us with large view? Maybe we can skip it for thumbnails.

Reply:
removed